texar/modules/encoders/bert_encoders.py

```diff
@@ -0,0 +1,356 @@
+    def default_hparams():
+        """Returns a dictionary of hyperparameters with default values.
+
+        * The encoder arch is determined by the constructor argument \
+        :attr:`pretrained_model_name` if it's specified. In this case, \
+        hparams are ignored.
+        * Otherwise, the encoder arch is determined by \
+        `hparams['pretrained_model_name']` if it's specified. All other \
+        configs in hparams are ignored.
+        * If the above two are `None`, the encoder arch is defined by \
+        the configs in hparams and weights are randomly initialized.
+
+        .. code-block:: python
+
+            {
+                'pretrained_model_name': 'bert-base-uncased',
+                'embed': {
+                    'dim': 768,
+                    'name': 'word_embeddings'
+                },
+                'vocab_size': 30522,
+                'segment_embed': {
+                    'dim': 768,
+                    'name': 'token_type_embeddings'
+                },
+                'type_vocab_size': 2,
+                'position_embed': {
+                    'dim': 768,
+                    'name': 'position_embeddings'
+                },
+                'position_size': 512,
+
+                'encoder': {
+                    'dim': 768,
+                    'embedding_dropout': 0.1,
+                    'multihead_attention': {
+                        'dropout_rate': 0.1,
+                        'name': 'self',
+                        'num_heads': 12,
+                        'num_units': 768,
+                        'output_dim': 768,
+                        'use_bias': True
+                    },
+                    'name': 'encoder',
+                    'num_blocks': 12,
+                    'poswise_feedforward': {
+                        'layers': [
+                            {   'kwargs': {
+                                    'activation': 'gelu',
+                                    'name': 'intermediate',
+                                    'units': 3072,
+                                    'use_bias': True
+                                },
+                                'type': 'Dense'
+                            },
+                            {   'kwargs': {'activation': None,
+                                'name': 'output',
+                                'units': 768,
+                                'use_bias': True
+                                },
+                                'type': 'Dense'
+                            }
+                        ]
+                    },
+                    'residual_dropout': 0.1,
+                    'use_bert_config': True
+                },
+                'hidden_size': 768,
+                'initializer': None,
+                'name': 'bert_encoder'
+            }
+
+
+
+        Here:
+
+        The default parameters are values for uncased BERT-Base model.
+
+
+        "pretrained_model_name" : str or None
```

Comment:
`texar.HParams` will automatically replace `None` with the default value (see [here](https://github.com/asyml/texar/blob/master/texar/hyperparams.py#L227)). Could you do a test and add to `bert_encoders_test.py`? 

If it does replace None with default, a workaround is perhaps to use a str like "customize" instead of None to indicate customized arch